File: src/fiefapp.svelte/src/models/builder.ts
import type { Building } from './building';
import type { IPerson } from './person';

interface IBuilder {
  building?: Building;
}

export class Builder implements IBuilder, IPerson {
  id: string;
  building?: Building;
  name: string;
  skill: number;
  resources: number;
  loyalty: number;
  age: number;

  constructor() {
    this.id = '';
    this.building = undefined;
    this.name = '';
    this.skill = 0;
    this.resources = 0;
    this.loyalty = 0;
    this.age = 0;
  }
}

File: src/fiefapp.entities/Builder.cs
using fiefapp.entities.BaseClasses;
using fiefapp.entities.Interfaces;
using System;
using System.Collections.Generic;

namespace fiefapp.entities
{
    public class Builder : Person, IBaseEntity, IFiefForeinKey
    {
        public Builder()
        {
            Id = Guid.NewGuid();
            FiefId = Guid.Empty;
            Notes = new HashSet<string>();
        }

        public override Guid Id { get; set; }
        public Guid FiefId { get; set; }
        public override string Name { get; set; }
        public override int Age { get; set; }
        public override int Resources { get; set; }
        public override int Loyalty { get; set; }
        public override int Skill { get; set; }
        public override ICollection<string> Notes { get; set; }
        public Guid? BuildingId { get; set; }
        public Building Building { get; set; }
    }
}
